src/components/recommender/extension-recommender.ts

```diff
@@ -0,0 +1,67 @@
+
+import * as vscode from 'vscode';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import { Kubectl } from '../../kubectl';
+
+interface Recommendations {
+    [crdApi: string]: string[];
+}
+
+async function recommendExtensionsToUser(tags: string[]): Promise<void> {
+    if (tags.length > 0) { // do not prompt if we cannot recommend anything
+        const answer = await vscode.window.showInformationMessage('Do you want to view recommended extensions for your Kubernetes cluster?', 'Don\'t ask again', 'Ignore', 'Search for Extensions');
+        if (answer === 'Search for Extensions') {
+            const searchValue = tags.map((it) => `@tag:${it}`).join(' ');
+            vscode.commands.executeCommand('workbench.extensions.search', searchValue);
+        } else if (answer === 'Don\'t ask again') {
+            await vscode.workspace.getConfiguration('extensions').update('ignoreRecommendations', true);
+        }
+    }
+}
+
+async function getCRDApiGroups(kubectl: Kubectl): Promise<string[]> {
+    const result = new Set<string>();
+    const invokeResult = await kubectl.invokeCommand('api-versions');
+    if (invokeResult.resultKind === 'exec-succeeded') {
+        try {
+            const res = invokeResult.stdout.split('\n');
+            res.forEach((i: any) => result.add(i));
+        } catch (err) {
+            // ignore
+        }
+    }
+    return Array.from(result);
+}
+
+async function readRecommendation(extPath: string): Promise<Recommendations> {
+    const fileContent = await fs.readFile(path.join(extPath, 'recommendation', 'recommended-tags.json'), 'utf8');
```

Comment:
not sure in this case, but isn't preferable to use the vscode.fs.readFile API to have better chance to have it working in a remote environment?

Reply:
Done